packages/ag-charts-community/src/chart/data/dataModel.ts

```diff
@@ -347,7 +342,14 @@ export class DataModel<
             }
         };
 
+        const keyScopes = new Set<ScopeId>();
+        const valueScopes = new Set<ScopeId>();
         for (const def of opts.props) {
+            const scopes = def.type === 'key' ? keyScopes : valueScopes;
+            if ('scopes' in def) {
```

Comment:
nit: I'd avoid `in` checks in general. Is it for TS narrowing?

Reply:
Yes, not all members have a `scopes` property, so this is the simplest way to differentiate that.

Reply:
Updated to use a `isScoped()` type-guard to improve readability.